### PR TITLE
Standardize public methods interface

### PR DESCRIPTION
**Pull request by jonatas**
I struggled a lot to remind the pattern as second or first parameter of all
methods and it was a mess between `match?`, `search`, `rewrite`, `replace`, etc.

Now all methods follow the same interface: the expression is the first argument.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -195,9 +195,8 @@ a single `Object`, like `a.b.c.d`. Its corresponding s-expression would be:
 You can also search using nested arrays with **pure values**, or **shortcuts** or
 **procs**:
 
-    Fast.match?(ast, [:send, [:send, '...'], :d]) # => true
-    Fast.match?(ast, [:send, [:send, '...'], :c]) # => false
-    Fast.match?(ast, [:send, [:send, [:send, '...'], :c], :d]) # => true
+    Fast.match? ast, [:send, [:send, '...'], :d]  # => true
+    Fast.match? ast, [:send, [:send, '...'], :c]  # => false
```
Comment (baweaver) [inline README.md]: In the previous examples you have `pattern, ast`, but in this one it's kept reversed?

Comment (jonatas) [reply README.md]: good catch 😎 Let me fix it.

File: docs/index.md [hunk 2]
```diff
@@ -64,9 +64,36 @@ The current version cover the following elements:
...
+
+It uses [astrolable](https://github.com/yujinakayama/astrolabe) gem behind the scenes:
+
+```ruby
+Fast.ast(Fast.ast("1").class
+=> Astrolabe::Node
+Fast.ast(Fast.ast("1").type
+=> :int
+Fast.ast(Fast.ast("1").children
+=> [1]
```
Comment (baweaver) [inline docs/index.md]: Missing closing parens on all three of these after `("1")` I think

File: docs/index.md [hunk 3]
```diff
@@ -352,3 +376,22 @@ all internal `.rb` files
 Fast.ruby_files_from(['lib']) # => ["lib/fast.rb"]
 ```
 
+## search_all
+
+Combines the [search_file](#search_file) with [ruby_files_from](#ruby_files_from)
+multiple locations and returns tuples with files and results.
+
+```ruby
+Fast.search_all("(def ast_from_file")
```
Comment (baweaver) [inline docs/index.md]: Missing paren

File: docs/index.md (hunk 2 above)
Comment (jonatas) [reply docs/index.md]: thanks a lot! something went wrong while copy and pasting 😄

File: docs/index.md (hunk 3 above)
Comment (jonatas) [reply docs/index.md]: it works in both ways but let me fix to not make it so confusing :D

**Pull request merged**